Libraries/Animated/src/AnimatedImplementation.js

```diff
@@ -1100,17 +1277,78 @@ class AnimatedProps extends Animated {
   }
 
   __detach(): void {
+    if (this.__isNative && this._animatedView) {
+      this.__disconnectAnimatedView();
+    }
     for (var key in this._props) {
       var value = this._props[key];
       if (value instanceof Animated) {
         value.__removeChild(this);
       }
     }
+    super.__detach();
   }
 
   update(): void {
     this._callback();
   }
+
+  __makeNative(): void {
```

Comment:
I don't see you handling the following case:

```
var a1 = Animated.Value(0);
var a2 = Animated.Value(1);

style={{transform: [{translateX: a1}, {translateY: a2}]}}

Animated.timing(a1, {useNativeDriver: true}).start();
Animated.timing(a2, {useNativeDriver: false}).start();
```

A value shouldn't be able to depend on both a native and non native animation. They would otherwise be updated by both native and js, race to update and use half an incomplete value.

What we should ensure is that the entire tree is either all native or all non-native, but not a mix of both.